src/xtgeo/surface/_regsurf_gridding.py

```diff
@@ -322,21 +322,26 @@ def surf_fill(self, fill_value=None):
         logger.info("Do fill... DONE")
 
 
-def smooth_median(self, iterations=1, width=1):
-    """Smooth a surface using a median filter.
+def _smooth(
+    self,
+    window_function: Callable[[np.ndarray], np.ndarray],
+    iterations: int = 1,
+    width: int = 1,
+) -> None:
+    """
+    Smooth a RegularSurface using a window function.
 
-    .. versionadded:: 2.1
```

Comment:
Might consider keeping the version added. I guess it won't show in the auto-generated docs being private now, but a comment like this is easier than git bisect. I guess we should discuss if we'll keep this convention going forward, though

Reply:
I think I removed it in this private function since it was kept in the public one, but I can add it back. Should I then also add versionchanged for this gaussian method?

Reply:
That's a good point. Maybe we should just leave them out for now. A bit hard to know what the next version will be anyway -- 3.5.0 or 3.4.3?

Reply:
That is a general dilemma with `version_added`. But since it is an API change we have to bump the second number (-> 3.5 in this case)